File: scalatra/src/main/scala/com/main/ApiServer.scala
package com.main

import javax.servlet.ServletContext

import com.controllers.ApiController
import org.scalatra.LifeCycle

class ApiServer extends LifeCycle {

  override def init(context: ServletContext) {
    context.mount(new ApiController, "/hello")
  }
}

object ApiServer {
  def main(args: Array[String]): Unit = {
    // Huh?
  }
}

File: sparkjava/src/main/java/com/main/ApiServer.java
package com.main;


import static spark.Spark.get;
import static spark.Spark.port;

import com.controllers.ApiController;


public class ApiServer {
    public static void main(String[] args) {
        port(System.getenv("PORT") == null ? 8080 : Integer.parseInt(System.getenv("PORT")));
        get("/hello", ApiController::hello);
    }
}
